agent/lib/kontena/service_pods/lifecycle_hook_manager.rb

```diff
@@ -16,14 +16,15 @@ def on_pre_start
         pre_start_hooks = hooks_for('pre_start')
         return if pre_start_hooks.size == 0
 
-        service_config = config_container(service_pod.dup)
-        service_config['Entrypoint'] = '/bin/sh'
-        service_config['Cmd'] = ['-c', 'sleep 600']
+        pre_pod = service_pod.dup
+        pre_pod.entrypoint = '/bin/sh'
+        pre_pod.cmd = ['-c', 'sleep 600']
+        service_config = config_container(pre_pod)
         service_container = create_container(service_config)
         service_container.start!
         pre_start_hooks.each do |hook|
           info "running pre_start hook: #{hook['cmd']}"
-          command = ['/bin/sh', '-c', hook['cmd']]
+          command = ['/w/w', '/bin/sh', '-c', hook['cmd']]
```

Comment:
The `/w/w` is only for the default `net: bridge` containers, but not `net: host`